psqlparse/nodes/parsenodes.py

```diff
@@ -284,3 +313,186 @@ def __init__(self, obj):
+    def __init__(self, obj):
+        self.arg = build_from_item(obj, 'arg')
+        self.type_name = build_from_item(obj, 'typeName')
+        self.location = obj.get('location')
+
+    def tables(self):
+        return set()
+
+
+class TypeName(Node):
+    def __init__(self, obj):
+        self.names = build_from_item(obj, 'names')
+        self.setof = obj.get('setof')
+        self.pct_type = obj.get('pct_type')
+        self.type_mods = build_from_item(obj, 'typmods')
+        self.type_mod = obj.get('typemod')
+        self.array_bounds = build_from_item(obj, 'arrayBounds')
+        self.location = obj.get('location')
+
+    def tables(self):
+        return set()
+
+
+class SortBy(Node):
+    DIR_DEFAULT = 0
+    DIR_ASC = 1
+    DIR_DESC = 2
+    DIR_USING = 3
+
+    NULLS_DEFAULT = 0
+    NULLS_FIRST = 1
+    NULLS_LAST = 2
+
+    def __init__(self, obj):
+        self.node = build_from_item(obj, 'node')
+        self.dir = obj.get('sortby_dir')
+        self.nulls = obj.get('sortby_nulls')
+        self.using = build_from_item(obj, 'useOp', 'Literal')
+        self.location = obj.get('location')
+
+    def tables(self):
+        return set()
+
+
+class WindowDef(Node):
+    FRAME_NONDEFAULT = 0x00001                #  any specified?
+    FRAME_RANGE = 0x00002                     #  RANGE behavior
+    FRAME_ROWS = 0x00004                      #  ROWS behavior
+    FRAME_BETWEEN = 0x00008                   #  BETWEEN given?
+    FRAME_START_UNBOUNDED_PRECEDING = 0x00010 #  start is U. P.
+    FRAME_END_UNBOUNDED_PRECEDING = 0x00020   #  (disallowed)
+    FRAME_START_UNBOUNDED_FOLLOWING = 0x00040 #  (disallowed)
+    FRAME_END_UNBOUNDED_FOLLOWING = 0x00080   #  end is U. F.
+    FRAME_START_CURRENT_ROW = 0x00100         #  start is C. R.
+    FRAME_END_CURRENT_ROW = 0x00200           #  end is C. R.
+    FRAME_START_VALUE_PRECEDING = 0x00400     #  start is V. P.
+    FRAME_END_VALUE_PRECEDING = 0x00800       #  end is V. P.
+    FRAME_START_VALUE_FOLLOWING = 0x01000     #  start is V. F.
+    FRAME_END_VALUE_FOLLOWING = 0x02000       #  end is V. F.
+
+    FRAME_START_VALUE = (FRAME_START_VALUE_PRECEDING
+                         | FRAME_START_VALUE_FOLLOWING)
+    FRAME_END_VALUE = (FRAME_END_VALUE_PRECEDING
+                       | FRAME_END_VALUE_FOLLOWING)
+    FRAME_DEFAULTS = (FRAME_RANGE
+                      | FRAME_START_UNBOUNDED_PRECEDING
+                      | FRAME_END_CURRENT_ROW)
+
+    def __init__(self, obj):
+        self.name = Name.from_string(obj.get('name'))
+        self.refname = Name.from_string(obj.get('refname'))
+        self.partition_clause = build_from_item(obj, 'partitionClause')
+        self.order_clause = build_from_item(obj, 'orderClause')
+        self.frame_options = obj.get('frameOptions')
+        self.start_offset = build_from_item(obj, 'startOffset')
+        self.end_offset = build_from_item(obj, 'endOffset')
+        self.location = obj.get('location')
+
+    def tables(self):
```

Comment:
Are you sure about this override? See https://github.com/alculquicondor/psqlparse/blob/master/psqlparse/nodes/nodes.py#L8 and don't implement the method if you are not sure about the node containing tables (for now). Do the same for any other Node.